### PR TITLE
[5.7] Add optimize and optimize:clear commands

### PR DESCRIPTION
**Pull request by vinkla**
Added `optimize` and `optimize:clear` commands to faster cache and clear cache.

As discussed in: https://github.com/laravel/internals/issues/171

Please let me know what you think and if there is anything I can improve.

### CODE REVIEW

**Comment by taylorotwell** [comment]
I'm not sure flushing the application cache is related to clearing the configuration and route caches, which you would do on every deployment. I don't think you would flush your entire application cache on every deployment.

File: src/Illuminate/Foundation/Console/OptimizeClearCommand.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $this->call('cache:clear');
+        $this->call('route:clear');
+        $this->call('view:clear');
+        $this->call('clear-compiled');
+
```
Comment (36864) [inline src/Illuminate/Foundation/Console/OptimizeClearCommand.php]: Shouldn't you also be calling `config:clear` here?

File: src/Illuminate/Foundation/Console/OptimizeCommand.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $this->call('cache:clear');
+        $this->call('config:cache');
+        $this->call('route:clear');
```
Comment (36864) [inline src/Illuminate/Foundation/Console/OptimizeCommand.php]: `route:cache` implicitly clears the route cache before creating it again.

Comment (vinkla) [reply src/Illuminate/Foundation/Console/OptimizeCommand.php]: Yeah, this has been removed in later commits.

File: src/Illuminate/Foundation/Console/OptimizeClearCommand.php (hunk 1 above)
Comment (vinkla) [reply src/Illuminate/Foundation/Console/OptimizeClearCommand.php]: I've submitted a follow-up PR: https://github.com/laravel/framework/pull/23473

**Comment by trevorgehman** [comment]
> I'm not sure flushing the application cache is related to clearing the configuration and route caches, which you would do on every deployment. I don't think you would flush your entire application cache on every deployment.

I'm curious why this was merged with `cache:clear` in place. Clearing the entire application cache is not an "optimization" that should be included without some consideration, IMO.

**Pull request merged**